django/core/cache/backends/db.py

```diff
@@ -54,10 +54,7 @@ def get_many(self, keys, version=None):
         if not keys:
             return {}
 
-        key_map = {}
-        for key in keys:
-            self.validate_key(key)
-            key_map[self.make_key(key, version)] = key
```

Comment:
I agree with Chris. We can add a test where `make_key()` will exceed the limit, e.g.
```python
    def test_invalid_with_version_key_length(self):
        # make_key() adds a version to the key and exceeds the limit.
        def func(key, *args):
            return key + ':1'
        
        old_func = cache.key_func
        cache.key_func = func
        
        key = 'a' * 249 
        expected_warning = (
            'Cache key will cause errors if used with memcached: '
            '%r (longer than %s)' % (func(key), 250)
        )   
            
        tests = [
            ('add', [key, 1]),
            ('get', [key]),
            ('set', [key, 1]),
            ('incr', [key]),
            ('decr', [key]),
            ('touch', [key]),
            ('delete', [key]),
            ('get_many', [[key, 'b']]),
            ('set_many', [{key: 1, 'b': 2}]),
            ('delete_many', [[key, 'b']]),
        ]               
        try:        
            for operation, args in tests:
                with self.subTest(operation=operation):
                    with self.assertWarns(CacheKeyWarning) as cm:
                        getattr(cache, operation)(*args)
                    self.assertEqual(str(cm.warning), expected_warning)
        finally:
            cache.key_func = old_func
```
It's a bugfix so we need regression tests.

Reply:
Sorry, I haven't had time to come back around to this. Feel free to push it to completion @felixxm.

I just have a couple of observations:
- The proposed change in the comment above will need to check that an exception is raised for the memcached backends rather than a warning.
- Length of key is not the only thing that makes it invalid. Please also add an assertion to check for the warning about invalid characters, e.g. space.
- I get the dislike of mocks, but surely the point is that we're checking that `.validate_key()` is called after `.make_key()`. A test like this makes that implicit rather than explicit. Either way, it's better than nothing and I'll let you choose.

Reply:
> * The proposed change in the comment above will need to check that an exception is raised for the memcached backends rather than a warning.

Yes, but it works now, so it's extra test coverage not a regression test and should go to a separate commit.

> * Length of key is not the only thing that makes it invalid. Please also add an assertion to check for the warning about invalid characters, e.g. space.

I don't think it's necessary to test `make_key()` that adds invalid characters, it would be a slight exaggeration. Here we test the existing issue.